src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -1165,17 +1173,36 @@ AS
     AND     SeriesInstanceUid = ISNULL(@seriesInstanceUid, SeriesInstanceUid)
     AND     SopInstanceUid = ISNULL(@sopInstanceUid, SopInstanceUid)
 
-    IF (@@ROWCOUNT = 0)
-    BEGIN
-        THROW 50404, 'Instance not found', 1;
-    END
+    IF @@ROWCOUNT = 0
+        THROW 50404, 'Instance not found', 1    
 
     -- Deleting tag errors
+    DECLARE @deletedErrors AS TABLE
+    (
+        TagKey BIGINT  
+    )
     DELETE XQTE
+        OUTPUT deleted.TagKey
+        INTO @deletedErrors
     FROM dbo.ExtendedQueryTagError as XQTE
     INNER JOIN @deletedInstances as d
     ON XQTE.Watermark = d.Watermark
 
+    -- Update error count
+    IF EXISTS (SELECT * FROM @deletedErrors)
+    BEGIN
+        -- Calculate error count
+        SELECT COUNT(1) as ErrorCount,
+               TagKey
+        INTO #errorCounts
```

Comment:
nit: maybe call this `#removed` or `#removedTag`

Reply:
used _deleted_ instead of _removed_, since this SP use _deletedInstances_ before. keep consistency.